projects/core/src/lib/map/helpers/layer-tree-node.helper.ts

```diff
@@ -104,7 +102,7 @@ export class LayerTreeNodeHelper {
     layerTreeNodes: ExtendedLayerTreeNodeModel[],
     id: string,
     transformer: (node: ExtendedLayerTreeNodeModel, children: T[]) => T | null): T | null {
-    const node = LayerTreeNodeHelper.findLayerTreeNode(layerTreeNodes, id);
+    const node = LayerTreeNodeHelper.findLayerTreeNode(layerTreeNodes, id) as ExtendedLayerTreeNodeModel;
```

Comment:
Prevent using `as ...` as much as possible, it disables typechecking completely. Here its better to rewrite the `findLayerTreeNode` method to something like:

`public static findLayerTreeNode<T extends LayerTreeNodeModel>(layerTreeNodes: T[], id: string) {`

and now TS will automatically infer that its a `ExtendedLayerTreeNodeModel` since that is what you are passing to the method.

Reply:
yeah, that was a shortcut on my side, and wasn't sure how `<T extends LayerTreeNodeModel>` would've been accepted. Fixed that now.